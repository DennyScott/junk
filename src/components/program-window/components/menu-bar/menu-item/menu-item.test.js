import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MenuItem from './menu-item';

const minProps = {
  name: 'test',
};

it('MenuItem renders correctly', () => {
  const wrapper = shallow(<MenuItem {...minProps} />);
  expect(wrapper.length).toBe(1);
});

it('MenuItem render matches snapshot', () => {
  const wrapper = shallow(<MenuItem {...minProps} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('MenuItem gets the first letter of the name to key', () => {
  const wrapper = shallow(<MenuItem {...minProps} />);
  const keyLetter = wrapper.find('.key-letter');
  expect(toJson(keyLetter.children())).toBe(minProps.name.charAt(0));
});

it('selected state should start on false', () => {
  const wrapper = shallow(<MenuItem {...minProps} />);
  expect(wrapper.state().selected).toBeFalsy();
});

it('selected state should turn to true when item is clicked', () => {
  const wrapper = shallow(<MenuItem {...minProps} />);
  wrapper.simulate('click');
  expect(wrapper.state().selected).toBeTruthy();
});
