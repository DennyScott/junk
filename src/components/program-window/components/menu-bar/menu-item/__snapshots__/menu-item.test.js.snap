// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem render matches snapshot 1`] = `
<span
  className="menu item false"
  onClick={[Function]}
>
  <span
    className="key-letter"
  >
    t
  </span>
  est
</span>
`;
